base-notebook/start.sh

```diff
@@ -11,140 +12,185 @@ else
-    if [[ "${NB_UID}" == "$(id -u jovyan 2>/dev/null)" && "${NB_GID}" == "$(id -g jovyan 2>/dev/null)" ]]; then
-        # User is not attempting to override user/group via environment
-        # variables, but they could still have overridden the uid/gid that
-        # container runs as. Check that the user has an entry in the passwd
-        # file and if not add an entry.
-        STATUS=0 && whoami &> /dev/null || STATUS=$? && true
-        if [[ "${STATUS}" != "0" ]]; then
-            if [[ -w /etc/passwd ]]; then
-                echo "Adding passwd file entry for $(id -u)"
-                sed -e "s/^jovyan:/nayvoj:/" /etc/passwd > /tmp/passwd
-                echo "jovyan:x:$(id -u):$(id -g):,,,:/home/jovyan:/bin/bash" >> /tmp/passwd
-                cat /tmp/passwd > /etc/passwd
-                rm /tmp/passwd
-            else
-                echo 'Container must be run with group "root" to update passwd file'
-            fi
-        fi
+    # Warn about misconfiguration of: desired username, user id, or group id
+    if [[ -n "${NB_USER}" && "${NB_USER}" != "$(id -un)" ]]; then
+        echo "WARNING: container must be started as root to change the desired user's name with NB_USER!"
+    fi
+    if [[ -n "${NB_UID}" && "${NB_UID}" != "$(id -u)" ]]; then
+        echo "WARNING: container must be started as root to change the desired user's id with NB_UID!"
+    fi
+    if [[ -n "${NB_GID}" && "${NB_GID}" != "$(id -g)" ]]; then
+        echo "WARNING: container must be started as root to change the desired user's group id with NB_GID!"
+    fi
 
-        # Warn if the user isn't going to be able to write files to ${HOME}.
-        if [[ ! -w /home/jovyan ]]; then
-            echo 'Container must be run with group "users" to update files'
-        fi
-    else
-        # Warn if looks like user want to override uid/gid but hasn't
-        # run the container as root.
-        if [[ -n "${NB_UID}" && "${NB_UID}" != "$(id -u)" ]]; then
-            echo "Container must be run as root to set NB_UID to ${NB_UID}"
-        fi
-        if [[ -n "${NB_GID}" && "${NB_GID}" != "$(id -g)" ]]; then
-            echo "Container must be run as root to set NB_GID to ${NB_GID}"
+    # Warn about misconfiguration of: granting sudo rights
+    if [[ "${GRANT_SUDO}" == "1" || "${GRANT_SUDO}" == 'yes' ]]; then
+        echo "WARNING: container must be started as root to grant sudo permissions!"
+    fi
+
+    # Attempt to ensure the user uid we currently run as has a named entry in
+    # the /etc/passwd file, as it avoids software crashing on hard assumptions
+    # on such entry. Writing to the /etc/passwd was allowed for the root group
+    # from the Dockerfile during build.
+    #
+    # ref: https://github.com/jupyter/docker-stacks/issues/552
+    if ! whoami &> /dev/null; then
+        if [[ -w /etc/passwd ]]; then
+            sed --in-place "s/^jovyan:/nayvoj:/" /etc/passwd
+            echo "Renamed old jovyan user to nayvoj (1000:100)"
```

Comment:
Why is it 1000:100 here?

Reply:
Are you asking why we are printing it, or why the user has UID of 1000 and GID of 100?

Reply:
This line occurs in the non-root branch, so UID cannot have been altered by `start.sh`. However, it could have been altered in a Dockerfile.

To do this properly, it seems to me like we should check whether a conflicting `${NB_USER:-jovyan}` entry exists in `/etc/passwd`, and if so rename, and then create the new user.

But for the moment, the current code looks consistent with the current behavior, so I think we should address this with a followup PR.

Reply:
87ad5ff should fix this for now.